src/VisualStudio/Core/Def/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -326,20 +323,6 @@ public void RunAnalyzers(IVsHierarchy? hierarchy)
                 otherProjectsForMultiTfmProject = ImmutableArray<Project>.Empty;
             }
 
-            bool isAnalysisDisabled;
-            if (project != null)
-            {
-                isAnalysisDisabled = _globalOptions.IsAnalysisDisabled(project.Language);
-            }
-            else
-            {
-                isAnalysisDisabled = true;
-                foreach (var language in solution.Projects.Select(p => p.Language).Distinct())
-                {
-                    isAnalysisDisabled = isAnalysisDisabled && _globalOptions.IsAnalysisDisabled(language);
-                }
-            }
```

Comment:
what happened here?

Reply:
This was the change done for fixing the issue that Sam pointed out when background code analysis is completely disabled and Run code analysis does nothing - both in LSP and non-LSP mode. https://github.com/dotnet/roslyn/commit/c892a8381692d62c559f20c5495310700cdae789 is the core fix, which mostly deletes a bunch of code and makes a small change in ForceAnalyzeAsync to handle this correctly.

Reply:
Perfect. Thanks!